flycheck.el

```diff
@@ -4792,6 +4814,16 @@ STRING
      To ignore the output of syntax checkers, try `null-device'
      first.
 
+`(force-suffix BODY SUFFIX...)'
+     Ensure generated file names end with one of the given
+     suffixes.  If not, append the first suffix in the list.  Use
+     this when the checker requires that file names on its
+     command line have a certain suffix (file extension).
+
+     BODY is evaluated as any other
+     `flycheck-substitute-argument' argument.  Affects `source',
+     `source-inplace', and `temporary-file-name'.
```

Comment:
I wonder if we need to support temporary-file-name — I don't think we have even a single checker in Flycheck that actually uses it.

Reply:
We don't have any.  However a Github search turns out a few hits on external checkers.  It's probably not an issue if we leave `temporary-file-name` out of this PR.